### PR TITLE
feat: add example terraform module to use workload identity in AKS

### PR DESCRIPTION
**Pull request by epomatti**
**Reason for Change**:

Figured it might be useful sharing it here.

If so, I'll archive [my repo](https://github.com/epomatti/azure-workload-identity-terraform) and reference this example directory.

**Requirements**

- [x] squashed commits
- [x] included documentation

**Issue Fixed**: n/a

**Please answer the following questions with yes/no**:

Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

- [x] yes, attributions will be welcomed if you add 🔥
- [ ] no

### CODE REVIEW

File: examples/terraform-aks/README.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+ℹ️ Modules are isolated for individual `apply` commands, following [this warning](https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs#stacking-with-managed-kubernetes-cluster-resources) from the Kubernetes provider.
+
+## Deployment Steps
+
+### 1 - Enable OIDC Issuer Preview
+
+Head over to this Microsoft Docs section: **[Register the `EnableOIDCIssuerPreview` feature flag](https://docs.microsoft.com/en-us/azure/aks/cluster-configuration#register-the-enableoidcissuerpreview-feature-flag)**
+
+Enable the feature (`az feature register`) and propagate it (`az provider register`).
```
[the remark this answers is not in the document]
Comment (epomatti) [reply examples/terraform-aks/README.md]: Makes sense, update here bbeb5858726fb37c8d30271e326ca06fd8882455

File: examples/terraform-aks/.gitignore [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+# Include override files you do wish to add to version control using negated pattern
+# !example_override.tf
+
+# Include tfplan files to ignore the plan output of command: terraform plan -out=tfplan
+# example: *tfplan*
+
+# Ignore CLI configuration files
+.terraformrc
+terraform.rc
```
[the remark this answers is not in the document]
Comment (epomatti) [reply examples/terraform-aks/.gitignore]: done: fe3396bb3ae1225a68b78f216f6597c784ccedff

never noticed but major gitignores have no eof new line. looks like an unspoken convention.
https://github.com/github/gitignore

File: examples/terraform-aks/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (epomatti) [reply examples/terraform-aks/README.md]: Migrated to topics so this probably looks ok now.

**Comment by sozercan** [comment]
@epomatti thanks for your work and PR! 🙏

Since Terraform templates are not part of the core functionality of this repo, I am afraid they'll get outdated over time. In addition to that, users might have the impression that they are maintained and tested by core maintainers and will be disappointed if they failed.

Would it be possible to add a docs update to refer to your repo as an example?

**Comment by epomatti** [comment]
@sozercan that sounds great.

I moved the minor code adjustments back again to my repo:

https://github.com/epomatti/azure-workload-identity-terraform

**Pull request closed without merging**